### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by 78fatemanazari**
## Description
In this project you will learn how to use a module inside your class. For this you will create a class `MyList` and a module `MyEnumerable`. Your module `MyEnumerable` will implement a subset of the functionality of [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html).

*IMPORTANT NOTE: Read **all** requirements before you start building your project.*

### General requirements

- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that you used correct ([GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Ruby requirements
- Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### Project requirements

- Check offical documentation about the following 3 methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html). Make sure that you understand what they are doing.
  - [description of #all? method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-all-3F)
  - [description of #any? method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-any-3F)
  - [description of #filter method](https://ruby-doc.org/core-3.0.0/Enumerable.html#method-i-filter)
- Create a class `MyList` that has an instance variable `@list`.
- In `MyList` implement a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.
- Create a module `MyEnumerable` that implements the following methods (they should have the same funcionality as methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html)):
  - `#all?`
  - `#any?`
  - `#filter`
- Each class and module should has a separate .rb file.
- Verify your solution:
```ruby
# Create our list
irb> list = MyList.new(1, 2, 3, 4)
=> #<MyList: @list=[1, 2, 3, 4]>

# Test #all?
irb> list.all? {|e| e < 5}
=> true
irb> list.all? {|e| e > 5}
=> false

# Test #any?
irb> list.any? {|e| e == 2}
=> true
irb> list.any? {|e| e == 5}
=> false

# Test #filter
irb> list.filter {|e| e.even?}
=> [2, 4]
```

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  include MyEnumerable
+
+  def initialize(list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (DeliceLydia) [inline my_list.rb]: - [ ] I suggest that method `each` method yields all members of `@list` in succession

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class MyList
+  require_relative 'my_enumerable'
+
+  include MyEnumerable
+
+  def initialize(list)
+    @list = list
+  end
```
Comment (DeliceLydia) [inline my_list.rb]: - [ ] Please make sure that the `#initialize` method accepts multiple arguments.
Screenshot 2023-10-19 at 18 46 38

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Created all required methods
- No linter errors


## Required Changes ♻️
_Check the comments under the review._


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @jcmunav63 and @78fatemanazari , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights
✔️ Linters passing
✔️ All methods implemented
## Optional suggestions

- [ ] Hello Team! 🚀 I'd like to emphasize the critical importance of adhering to the project requirements and guidelines. In this specific project and milestone, we explicitly specified the use of GitHub flow, which involves creating pull requests from the feature branch to the main branch. 🌟

- Moving forward, it's essential to be diligent in following these guidelines. 📌 Doing so ensures a smooth workflow and collaboration, and it helps maintain consistency in our project management practices. Let's keep striving for excellence and maintain a strong commitment to adhering to the defined processes. 🎯 Thank you for your cooperation! 🙏

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**